Octokit.Tests.Integration/Clients/RepositoryBranchesClientTests.cs

```diff
@@ -543,4 +544,999 @@ public TheDeleteBranchProtectionMethod()
+
+                Assert.NotNull(deleted);
+                Assert.True(deleted.Contains("test"));
+            }
+        }
+
+        [IntegrationTest]
+        public async Task DeletesRequiredStatusChecksContextsForOrgRepo()
+        {
+            using (var context = await _github.CreateOrganizationRepositoryWithProtectedBranch())
+            {
+                var repoOwner = context.RepositoryContext.RepositoryOwner;
+                var repoName = context.RepositoryContext.RepositoryName;
+                var contextsToRemove = new List<string>() { "build" };
+                var deleted = await _client.DeleteRequiredStatusChecksContexts(repoOwner, repoName, "master", contextsToRemove);
+
+                Assert.NotNull(deleted);
+                Assert.True(deleted.Contains("test"));
+            }
+        }
+
+        [IntegrationTest]
+        public async Task DeletesRequiredStatusChecksContextsForOrgRepoWithRepositoryId()
+        {
+            using (var context = await _github.CreateOrganizationRepositoryWithProtectedBranch())
+            {
+                var repoId = context.RepositoryContext.RepositoryId;
+                var contextsToRemove = new List<string>() { "build" };
+                var deleted = await _client.DeleteRequiredStatusChecksContexts(repoId, "master", contextsToRemove);
+
+                Assert.NotNull(deleted);
+                Assert.True(deleted.Contains("test"));
+            }
+        }
+    }
+
+    public class TheGetProtectedBranchRestrictionsMethod : IDisposable
+    {
+        IRepositoryBranchesClient _client;
+        OrganizationRepositoryWithTeamContext _orgRepoContext;
+
+        public TheGetProtectedBranchRestrictionsMethod()
+        {
+            var github = Helper.GetAuthenticatedClient();
+            _client = github.Repository.Branch;
+
+            _orgRepoContext = github.CreateOrganizationRepositoryWithProtectedBranch().Result;
+        }
+
+        [IntegrationTest]
+        public async Task GetsRequirProtectedBranchRestrictionsForOrgRepo()
+        {
+            var repoOwner = _orgRepoContext.RepositoryContext.RepositoryOwner;
+            var repoName = _orgRepoContext.RepositoryContext.RepositoryName;
+            var restrictions = await _client.GetProtectedBranchRestrictions(repoOwner, repoName, "master");
+
+            Assert.NotNull(restrictions);
+            Assert.NotNull(restrictions.Teams);
+            Assert.NotNull(restrictions.Users);
+            Assert.Equal(1, restrictions.Teams.Count);
+            Assert.Equal(0, restrictions.Users.Count);
+        }
+
+        [IntegrationTest]
+        public async Task GetsProtectedBranchRestrictionsForOrgRepoWithRepositoryId()
+        {
+            var repoId = _orgRepoContext.RepositoryContext.RepositoryId;
+            var restrictions = await _client.GetProtectedBranchRestrictions(repoId, "master");
+
+            Assert.NotNull(restrictions);
+            Assert.NotNull(restrictions.Teams);
```

Comment:
since you are accessing `restriction.Users.Count` anyway there isnt a need to assert that `restrictions` and `restrictions.Users` arent `null`.  The `Count` assert will fail if any of those were `null` anyway